spec/knapsack_pro/adapters/rspec_adapter_spec.rb

```diff
@@ -305,6 +305,15 @@
         expect(subject).to eq('')
       end
     end
+
+    context "when id does not end in .feature (nor _spec.rb)" do
```

Comment:
I'm wondering if this description is a little confusing.

Reply:
Ignore my previous comment.

We could just clarify the description. For this new test we specifically want to test the .feature extension detection works. 

```suggestion
    context "when id does not end with a valid extension AND file_path has .feature extension" do
```

We could also update old descriptions like L275

```ruby
context "when id, file_path, and example_group's file_path do not end in a valid extension"
```

Reply:
To me, it is confusing that the description says `context "when id does not end in .feature (nor _spec.rb)"`. So it sounds like .feature and _spec.rb is not important.
But then inside of `it` we expect `file_path` to end with `.feature`.

Also the above test is confusing with L255 test:

```ruby
    context "when id does not end in _spec.rb" do
      it "returns the file_path" do
        allow(current_example).to receive(:id).and_return("./foo.rb")
        allow(current_example).to receive(:metadata).and_return(file_path: "./foo_spec.rb")
        expect(subject).to eq('./foo_spec.rb')
      end
    end
```

Confusing to me is that both descriptions mean the same and I don't see how they are different from each other. You need to look at the source code of the test case to find a difference.

* context "when id does not end in .feature (nor _spec.rb)"
* context "when id does not end in _spec.rb"

Possible solution:
Instead of saying in a negative way, `id does not end with something`. We could say it in a positive way. So that 

`when id ends with an invalid extension AND file_path ends with .feature`.
then we expect to get a file_path with .feature inside of `it` test case.

I'd do something like this:

```ruby
context 'when id ends with an invalid extension AND file_path ends with .feature' do
  it 'returns a file_path with .feature extension' do
  end
end
```